shells/android/java/arcs/storage/service/IStorageServiceCallback.aidl

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2019 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.storage.service;
+
+import arcs.storage.parcelables.ParcelableModelUpdate;
+import arcs.storage.parcelables.ParcelableOperations;
+import arcs.storage.parcelables.ParcelableSyncRequest;
+import arcs.storage.service.IResultCallback;
+
+/** Variant of ProxyCallback intended for StorageService communications. */
+interface IStorageServiceCallback {
+    /** Handles an incoming SyncRequest. */
+    void onSyncRequest(in ParcelableSyncRequest req, IResultCallback resultCallback);
```

Comment:
Very minor nit: can we call this `request` instead of `req`? No need to be so brief.

Reply:
Yep, I should have mentioned in the PR description that all of the callbacks and non-parcelable aidl files are tentative & subject to change. They are mostly here in this PR to serve as a way to verify that I can build AIDL.